fairlearn/adversarial/_adversarial_mitigation.py

```diff
@@ -0,0 +1,822 @@
+                    )
+                )
+            else:
+                for cb in self.callbacks:
+                    if not callable(cb):
+                        raise ValueError(
+                            _KWARG_ERROR_MESSAGE.format(
+                                "callbacks", "a callable or list of callables"
+                            )
+                        )
+
+        # NOTE: inferring distribution type should happen before transforming
+        read_kw = (
+            lambda data, kw_or_func: _get_type(data, kw_or_func)
+            if isinstance(kw_or_func, str)
+            else kw_or_func
+        )
+
+        self.predictor_loss_ = read_kw(Y, self.predictor_loss)
+        self.adversary_loss_ = read_kw(Z, self.adversary_loss)
+        self.predictor_function_ = read_kw(Y, self.predictor_function)
+
+        kws = ["binary", "category", "continuous"]
+        for kw, kwname in (
+            (self.predictor_loss_, "predictor_loss"),
+            (self.adversary_loss_, "adversary_loss"),
+            (self.predictor_function_, "predictor_function"),
+        ):
+            # Some keywords can also be a callable instead of a str.
+            if not ((isinstance(kw, str) and kw in kws) or callable(kw)):
+                raise ValueError(
+                    _KWARG_ERROR_MESSAGE.format(
+                        kwname,
+                        (
+                            "'auto', 'binary', 'category', 'continuous', "
+                            + "or a callable",
+                        ),
+                    )
+                )
+
+        for kw, kwname in (
+            (self.y_transform, "y_transform"),
+            (self.z_transform, "z_transform"),
+        ):
+            if not (
+                issubclass(type(kw), TransformerMixin)
+                and hasattr(kw, "fit")
+                and hasattr(kw, "transform")
+            ):
+                raise ValueError(
+                    _KWARG_ERROR_MESSAGE.format(
+                        kwname,
+                        "a sklearn Transformer (subclass TransformerMixin)",
+                    )
+                )
+
+        self.y_transform_ = self.y_transform.fit(Y)
+        self.z_transform_ = self.z_transform.fit(Z)
+
+        if self.cuda and not isinstance(self.cuda, str):
+            raise ValueError(
+                _KWARG_ERROR_MESSAGE.format(
+                    "cuda",
+                    'None, or the device to use (such as "cuda:0")',
+                )
+            )
+
+        self.random_state_ = check_random_state(self.random_state)
+
+        self._set_predictor_function()
+
+        # Initialize backend
+        # here, losses and optimizers are also set up.
+        self.backendEngine_ = self.backend_(self, X, Y, Z)
+
+        # Sklearn-parameters
+        self.n_features_in_ = X.shape[1]
+        self.n_features_out_ = self.y_transform_.n_features_in_
+
+    def fit(self, X, y, *, sensitive_features=None):
```

Comment:
What will happen if nothing is passed for `sensitive_features` ?

Reply:
That is a good question, I should check these cases.

Reply:
Was raising a warning, but changed it to an Exception